vllm_spyre/v1/worker/spyre_model_runner.py

```diff
@@ -1740,3 +1741,565 @@ def execute_model(
+                                      dtype=torch.int64,
+                                      device=self.device)
+        input_positions_np = input_positions.numpy()
+
+        # create block table tensor
+        blocks = [0] * (left_padding // self.block_size) + list(
+            self.req_ids2blocks[req_id])
+        block_end = math.ceil((chunk_start + chunk_size) / self.block_size)
+        block_table = torch.tensor(blocks[:block_end],
+                                   dtype=torch.int64).unsqueeze(0)
+
+        slot_mapping = []
+        for i in range(chunk_blocks):
+            block = block_table[0][-chunk_blocks + i]
+            slot_mapping += list(
+                range(block * self.block_size,
+                      block * self.block_size + self.block_size))
+        slot_mapping = torch.tensor(slot_mapping,
+                                    device=self.device,
+                                    dtype=torch.int64).unsqueeze(0)
+
+        left_pad_blocks_offset = 0 if left_padding == 0 \
+            else chunk_size - left_padding
+
+        if prompt_len < chunk_size and num_computed_tokens == 0:
+            # Case I - Prompt fits in a single chunk
+            input_tokens_np[left_padding:left_padding +
+                            chunk_end] = prompt_token_ids
+            input_positions_np[left_padding:left_padding +
+                               chunk_end] = range(prompt_len)
+        elif left_padding > 0 and num_computed_tokens == 0:
+            # Case II - First chunk but it contains some left blocks at the left
+            input_tokens_np[
+                left_padding:left_padding+left_pad_blocks_offset] = \
+                    prompt_token_ids[:left_pad_blocks_offset]
+            input_positions_np[
+                left_padding:left_padding+left_pad_blocks_offset] = \
+                    range(left_pad_blocks_offset)
+        else:
+            # The `token_idx` is the offset of the chunking in the prompt,
+            # because we can have left pad blocks, we may not slice it with
+            # the exact size of the chunk.
+            if left_padding == 0:
+                # Case III
+                # No left padding, it will start with full chunks
+                token_idx = chunk_i * chunk_size
+            else:
+                # Case II - remaining chunks
+                # The `left_pad_blocks_offset` is the piece of the first
+                # chunk that needed to be incomplete to compensate left
+                # padding. Sum it with the remaining chunks count to get
+                # where to slice the prompt
+                token_idx = left_pad_blocks_offset + (chunk_i - 1) * chunk_size
+
+            input_tokens_np[:chunk_end - token_idx] = (
+                prompt_token_ids[token_idx:min(token_idx +
+                                               chunk_size, prompt_len)])
+            input_positions_np[:chunk_end - token_idx] = range(
+                token_idx, chunk_end)
```

Comment:
there is something wrong with `chunk_end`, I just tried with a long prompt that spans 3 chunks: `prompt_len` = 260, `chunk_size` = 128. 
For the second chunk, we have `chunk_end = 256`, and this raises error because now we go over the size of `input_tokens` which is 192 so `range(token_idx, chunk_end)` produces more values than the tensor can accept. Also `chunk_end - token_idx` produces value that doesn't make sense imo.